opensrp-reveal/src/main/java/org/smartregister/reveal/sync/LocationTaskIntentService.java

```diff
@@ -42,17 +42,22 @@ private void doSync() {
         LocationServiceHelper locationServiceHelper = LocationServiceHelper.getInstance();
         TaskServiceHelper taskServiceHelper = TaskServiceHelper.getInstance();
 
+
         List<Location> syncedStructures = locationServiceHelper.fetchLocationsStructures();
         List<Task> synchedTasks = taskServiceHelper.syncTasks();
 
-
         if (hasChangesInCurrentOperationalArea(syncedStructures, synchedTasks)) {
             Intent intent = new Intent(STRUCTURE_TASK_SYNCHED);
             LocalBroadcastManager.getInstance(getApplicationContext()).sendBroadcast(intent);
         }
 
         clientProcessEvents(extractStructureIds(syncedStructures, synchedTasks));
 
+        if (!org.smartregister.util.Utils.isEmptyCollection(syncedStructures)
+                || !org.smartregister.util.Utils.isEmptyCollection(syncedStructures)) {
+            doSync();
```

Comment:
Does this have a risk of causing an infinite loop if for some reason one of the lists is always empty?

Reply:
This would be invoked in case either of the list is not empty which mean there were structures or tasks that were synced. 

When both lists are empty the method returns

Reply:
Okay.